hack/generated/pkg/reconcilers/azure_deployment_reconciler.go

```diff
@@ -560,7 +540,40 @@ func (r *AzureDeploymentReconciler) CreateDeployment(ctx context.Context) (ctrl.
 	return result, err
 }
 
-// TODO: There's a bit too much duplicated code between this and create deployment -- should be a good way to combine them?
+func (r *AzureDeploymentReconciler) handleDeploymentFinished(ctx context.Context, deployment *armclient.Deployment) (ctrl.Result, error) {
+	var status genruntime.FromARMConverter
+	if deployment.IsSuccessful() {
+		// TODO: There's some overlap here with what Update does
+		if len(deployment.Properties.OutputResources) == 0 {
+			return ctrl.Result{}, errors.Errorf("template deployment didn't have any output resources")
+		}
+
+		resourceID, err := deployment.ResourceID()
+		if err != nil {
+			return ctrl.Result{}, errors.Wrap(err, "getting resource ID from resource")
+		}
+
+		status, _, err = r.getStatus(ctx, resourceID)
+		if err != nil {
+			return ctrl.Result{}, errors.Wrap(err, "getting status from ARM")
+		}
+	}
+
+	err := r.Update(deployment, status)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "updating obj")
+	}
+
+	err = r.CommitUpdate(ctx)
+	if err != nil {
+		// This is a superfluous error as per https://github.com/kubernetes-sigs/controller-runtime/issues/377
+		// The correct handling is just to ignore it and we will get an event shortly with the updated version to patch
```

Comment:
Should we be treating _all_ errors as superfluous, or should we be checking in case an unexpected kind of error comes back?

Reply:
This comment applies specifically to `NotFound` which is dealt with via `client.IgnoreNotFound(err)` in the return below. The comment is a line above that because it's long,

I could change "this" to "NotFound" maybe?

Reply:
Changed it to:
> NotFound is a superfluous error as per https://github.com/kubernetes-sigs/controller-runtime/issues/377

Reply:
There were some other occurrences of this comment and I changed them too

Reply:
Err, ahh, my comment was more that currently _every_ error returned from `r.CommitUpdate()` is treated as superfluous regardless of the details, and I was asking if that was correct.

So I was asking about the couple of lines ABOVE the comment, not the lines below it.

Reply:
Having pulled the code and looked at in detail, it does the right thing - but it's an odd idiom and easy to misread.